docs/development/designs/automationGo.md

```diff
@@ -0,0 +1,49 @@
+# Problem
+
+Maintaining unsupported versions of Kubernetes and by extention Golang/Go requires large amounts of monitoring, evaluation, occassional deep dives into the functionality and differences in versions, and large amounts of repetitive searchable documentation for the Open-Source Community and transparency.
+
+Specifically for patching golang, we are required to monitor their announcements for vulnerabilities, open our own CVE issues for backports that are consistent to allow for easily searchable auditing and transparency for the open-source community, attempt backporting the fixes, and when the backport doesn't apply cleanly, discover the differences in versions to make the backport apply, if applicable.
+
+While the last task will require effort by the team, the rest of the task can be automated to provide clear and consistent [unknown phrasing/word here]. An automated solution would build a consistant audit trail for which CVEs have been patched or why they weren't, any public discussion on the decision process, and many other data points that could be of interest.
+
+# Solution Overview
+
+I propose creating a go application for patching. Starting with Golang and potentially extending the functionality to Kubernetes as the workflow should be similar.
+
+## Golang Patching Runbook
+
+1. Golang release CVE announcement
```

Comment:
any thoughts about the approach for determining if a Go release has happened? There are a few options wondering what you had in mine.

Reply:
In a brief search they are fairly consistent with there release issues. So,

* A daily prow job to check for new open issues in the golang/go repo. Since the issues usually have a link to the fix used for creating the cherry pick, we would have that as well. This would require human monitoring on the Golang announcements, because there is potential, if they don't follow the standard or our search pattern, we miss a CVE. 

* The other option, that would be a bit more technical, would be finding a way to parse the announcements for the CVE, then finding the link to the commit for cherrypicking. As long as our tests and logs reported failures to connect to the announcements, we would in theory catch all CVEs. 

Option 2 could be an advancement, but I think for initial development the option 1 would allow us to have a design that would effectively work and open time for advancements and other projects.